lib/linux.ts

```diff
@@ -0,0 +1,205 @@
+import debugFactory from 'debug'
+import { linuxList } from './linux-list'
+import { Poller } from './poller'
+import { unixRead } from './unix-read'
+import { unixWrite } from './unix-write'
+import { BindingInterface, OpenOptions, PortStatus, SetOptions, UpdateOptions } from './binding-interface'
+import { asyncOpen, asyncClose, asyncUpdate, asyncSet, asyncGet, asyncGetBaudRate, asyncFlush, asyncDrain } from './load-bindings'
+
+const debug = debugFactory('serialport/bindings-cpp')
+
+export interface LinuxBindingOptions {
+  /** see [`man termios`](http://linux.die.net/man/3/termios) defaults to 1 */
+  vmin?: number
+  /** see [`man termios`](http://linux.die.net/man/3/termios) defaults to 0 */
+  vtime?: number
+  // * @param {Boolean} [options.lowLatency=false] flag for lowLatency mode on Linux
+}
+
+export interface LinuxPortStatus extends PortStatus {
+  lowLatency: boolean
+}
+
+export interface LinuxSetOptions extends SetOptions {
+  /** Low latency mode */
+  lowLatency?: boolean
+}
+
+/**
+ * The linux binding layer
+ */
+export class LinuxBinding extends BindingInterface {
+  bindingOptions: LinuxBindingOptions
+  fd: null | number
+  writeOperation: Promise<void> | null
+  openOptions: (LinuxBindingOptions & OpenOptions) | null
+  poller: Poller | null
+
+  static list() {
+    debug('list')
+    return linuxList()
```

Comment:
Is this aysnc? Based on the `linux-list` file, this returns a promise. I would have expected it to match the signature of the `DarwinBinding.list`